src/features/generic_events/aggregate/index.js

```diff
@@ -90,14 +96,14 @@ export class Aggregate extends AggregateBase {
     }
 
     const eventAttributes = {
-      /** Agent-level custom attributes */
-      ...(getInfo(this.agentIdentifier).jsAttributes || {}),
       /** Common attributes shared on all generic events */
       referrerUrl: this.referrerUrl,
       currentUrl: cleanURL('' + location),
       pageUrl: cleanURL(getRuntime(this.agentIdentifier).origin),
       /** Event-specific attributes take precedence over everything else */
-      ...obj
+      ...obj,
+      /** Agent-level custom attributes */
+      ...(getInfo(this.agentIdentifier).jsAttributes || {})
```

Comment:
this is the behavior i mentioned that was marks as "to be preserved" in our slack chat, since the current implementation does not behave the way you described

Reply:
need to keep attribute precedence:

1. hardcoded properties, then
2. jsattributes || {}, then
3. local "attributes" passed into addPageAction